### PR TITLE
Allow python local imports if they solve a cyclic import problem

### PR DESCRIPTION
**Pull request by ericvergnaud**
Our current CI blocks merge of PRs if added python code contains comments such as:

`# pylint: disable=cyclic-import, import-outside-toplevel`

Local imports are discouraged, but are the most effective way of solving cyclic imports.

This PR solves the issue.

### CODE REVIEW

**Comment by ericvergnaud** [comment]
@gueniai you'll have to force merge of the PR because the test file contains examples of patterns we're not letting pass

File: tests/unit/no_cheat.py [hunk 1]
```diff
@@ -14,9 +18,11 @@ def no_cheat(diff_text: str) -> str:
         idx = line.find(DISABLE_TAG)
         if idx < 0:
             continue
-        codes = line[idx + len(DISABLE_TAG) :].split(',')
+        codes = {_strip_code(code) for code in line[idx + len(DISABLE_TAG) :].split(',')}
+        allowed_local_cyclic_imports = {'cyclic-import', 'import-outside-toplevel'}
+        if len(codes.intersection(allowed_local_cyclic_imports)) == 2:
```
Comment (vil1) [inline tests/unit/no_cheat.py:23]:
```suggestion
        if len(codes.intersection(allowed_local_cyclic_imports)) == len(allowed_local_cyclic_imports):
```
to make future changes easier

**Comment by sundarshankar89** [comment]
@ericvergnaud I would like to hold on to this change till we absolutely need it. I think there is an alternate way where you introduce your local import. 

Thoughts?

**Comment by ericvergnaud** [comment]
> @ericvergnaud I would like to hold on to this change till we absolutely need it. I think there is an alternate way where you introduce your local import.
> 
> Thoughts?

As discussed, this PR does not encourage local imports, rather it tolerates them in very specific use cases. PR reviews can still block PRs.

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Review by gueniai — APPROVED** [review]
LGTM

**Pull request merged**